### PR TITLE
Melf/add ruff 2024

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

Please summarise the changes.

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: pytket/extensions/aqt/multi_zone_architecture/circuit_routing/partition_routing.py [hunk 1]
```diff
@@ -65,9 +65,8 @@ def get_routed_circuit(self) -> MultiZoneCircuit:
         )
         for old_place, new_place in self.placement_generator(depth_list):
             if self._settings.debug_level > 0:
-                print("-------")
                 for zone in range(self._arch.n_zones):
-                    changes_str = ", ".join(
+                    ", ".join(
```
Comment (cqc-alec) [inline pytket/extensions/aqt/multi_zone_architecture/circuit_routing/partition_routing.py:69]: This string now unused -- delete?

Comment (cqc-alec) [reply pytket/extensions/aqt/multi_zone_architecture/circuit_routing/partition_routing.py:69]: I am not sure why we are deleting all the `print()` calls. They are presumably for debugging.

**Review by cqc-alec — COMMENTED** [review]
Is it intended to remove all debug and logging? If we want to get rind of `print()` statements entirely then maybe these should use [logging](https://docs.python.org/3/library/logging.html) instead?

File: pytket/extensions/aqt/multi_zone_architecture/circuit_routing/partition_routing.py (hunk 1 above)
Comment (cqc-melf) [reply pytket/extensions/aqt/multi_zone_architecture/circuit_routing/partition_routing.py]: Ruff suggested that we should remove the print / replace it with some special log. I think in our case it is fine to have the prints. I have added them back in.

**Pull request merged**